rca/account_management/views.py

```diff
@@ -0,0 +1,139 @@
+        student_group = Group.objects.get(name="Students")
+        student_user = User.objects.create_user(
+            username=form.cleaned_data["username"],
+            email=form.cleaned_data["email"],
+            first_name=form.cleaned_data["first_name"],
+            last_name=form.cleaned_data["last_name"],
+        )
+        # Add the student group
+        student_user.groups.add(student_group)
+        student_user.set_unusable_password()
+        student_user.save()
+
+        # return here if we aren't creating a page for the student
+        if not form.cleaned_data["create_student_page"]:
+            messages.success(
+                self.request,
+                f"The Student account for {student_user} has been created.",
+            )
+            return data
+
+        # Is there a student index page we can create a student page under?
+        student_index = StudentIndexPage.objects.first()
+        if not student_index:
+            messages.warning(
+                self.request,
+                f"Failed to create a Student Page matching for the new user: {student_user}, "
+                "a Parent Student Index Page is required.",
+            )
+            logger.info(
+                "There is not parent Student Index page to create a Student Page under, "
+                f"so one has not been created for user: {student_user}"
+            )
+            return data
+
+        # If a student page already exist for this user, we don't want to create one.
+        student_page = StudentPage.objects.filter(
+            first_name=form.cleaned_data["first_name"],
+            last_name=form.cleaned_data["last_name"],
+        ).exists()
+
+        if student_page:
+            messages.success(
+                self.request,
+                f"The Student account for {student_user} has been created.",
+            )
+            messages.warning(
+                self.request,
+                f"Failed to create a Student Page matching for the new user: {student_user}, "
+                "a Student Page already already exists",
+            )
+            logger.info(
+                "A Student User has been created, but a StudentPage matching "
+                f"the users name already exists: {student_page}, so one has not been created"
+            )
+        else:
+            student_index.add_child(
+                instance=StudentPage(
+                    first_name=form.cleaned_data["first_name"],
+                    last_name=form.cleaned_data["last_name"],
+                    title=f"{form.cleaned_data['first_name']} {form.cleaned_data['last_name']}",
+                    student_user_account=student_user,
+                    live=False,
+                )
+            )
+
+            # Generate a password reset link to be emailed to the user.
+            password_reset_url = self.request.build_absolute_uri(
+                get_set_password_url(student_user)
+            )
```

Comment:
This builds me out the correct URL to send to the user, however I'm struggling to see how to work a redirect to edit the studentpage. I can't do this with a param after the reset link as the user is asked to login using their new password.

So I think I might need to be looking at trying to hook into the login view, and if the user has the role 'Student', look for their page and redirect them. But I'm not sure where to start with this :/

Reply:
I agree that overriding [`LoginView`](https://github.com/wagtail/wagtail/blob/dede38be65f2b61726b99ff935a154e91662fc35/wagtail/admin/views/account.py#L289) would be a good way of going about this - specifically with a custom `get_success_url`. I think for this approach you could do the following:

1. Subclass `LoginView` and customise `get_success_url` as required
2. Override the `wagtailcore_login` path in the site's _urls.py_ to use your new view, instead of this one declared in `wagtail.core.urls`:

```
    path(
        '_util/login/',
        auth_views.LoginView.as_view(template_name=WAGTAIL_FRONTEND_LOGIN_TEMPLATE),
        name='wagtailcore_login'),
```

(If for any reason you're not able to/didn't want to override that path directly, you can set up a new path for the custom view, and set `WAGTAIL_FRONTEND_LOGIN_URL` to use it instead.)

Alternatively, you could look at overriding `PasswordResetCompleteView` somehow, so that the link to log in contains an appropriate parameter for redirecting, but I think I'd prefer the first approach.

Reply:
Thanks @alxbridge, I think I've found a way to override the LoginView here, so I'll get this code reviewed